DQMServices/FwkIO/plugins/DQMRootOutputModule.cc

```diff
@@ -293,14 +299,18 @@ DQMRootOutputModule::DQMRootOutputModule(edm::ParameterSet const& pset)
       m_presentHistoryIndex(0),
       m_filterOnRun(pset.getUntrackedParameter<unsigned int>("filterOnRun")),
       m_fullNameBufferPtr(&m_fullNameBuffer),
-      m_indicesTree(nullptr) {
+      m_indicesTree(nullptr),
+      m_getterOfProductsLumi(edm::TypeMatch(), this, edm::InLumi),
+      m_getterOfProductsRun(edm::TypeMatch(), this, edm::InRun) {
   // Declare dependencies for all Lumi and Run tokens here. In
   // principle could use the keep statements, but then DQMToken would
   // have to be made persistent (transient products are ignored),
   // which would lead to a need to (finally) remove underscores from
   // DQM module labels.
```

Comment:
How about adding something along "This is needed to support unscheduled DQM modules now that non-consumed EDProducers are deleted from the job at beginJob"? (to make the rationale more clear for the next time)

Reply:
I added that comment verbatim. Thanks